lib/command/src/Obelisk/Command/Thunk.hs

```diff
@@ -775,6 +778,17 @@ getThunkPtr' checkClean thunkDir upstream = do
         }
       }
 
+isBranchPushed :: MonadObelisk m => FilePath -> Text -> Text -> m Bool
+isBranchPushed thunkDir hd rm = do
+  hasBranch <- not . null . T.lines <$> readGitProcess thunkDir ["ls-remote", "--heads", T.unpack rm, T.unpack hd]
```

Comment:
This fixes an unfriendly error when a branch does not even exist in a remote, e.g.
```
RawCommand "nix-shell" ["-p","git","--run","'git' '-C' 'reflex-platform' 'merge-base' 'f4c17b91e6104830667ded50e950dd08f5e79bfe' 'remotes/origin/fix-ios-simulator'"] failed with exit code: 128
```